tests/v2/validation/networking/connectivity/network_policy.go

```diff
@@ -77,3 +101,302 @@ func curlCommand(client *rancher.Client, clusterID string, url string) (string,
+	if err != nil {
+		return false, err
+	}
+
+	provisioningClusterID, err := clusters.GetV1ProvisioningClusterByName(client, client.RancherConfig.ClusterName)
+	if err != nil {
+		return false, err
+	}
+
+	cluster, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ByID(provisioningClusterID)
+	if err != nil {
+		return false, err
+	}
+
+	newCluster := &provv1.Cluster{}
+	err = steveV1.ConvertToK8sType(cluster, newCluster)
+	if err != nil {
+		return false, err
+	}
+
+	if strings.Contains(newCluster.Spec.KubernetesVersion, "k3s") {
+		wranglerContext, err := client.WranglerContext.DownStreamClusterWranglerContext(clusterID)
+		if err != nil {
+			return false, err
+		}
+
+		latestDaemonset, err := wranglerContext.Apps.DaemonSet().List("kube-system", metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		for _, item := range latestDaemonset.Items {
+			if strings.Contains(item.Name, "svclb-traefik") {
+				return true, nil
+			}
+		}
+
+		return false, nil
+	}
+
+	_, err = catalogClient.Apps(kubeSystemNamespace).Get(context.TODO(), cloudControllerManager, metav1.GetOptions{})
+	if !strings.Contains(newCluster.Spec.KubernetesVersion, "k3s") && err != nil && strings.Contains(err.Error(), "not found") {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func isNodePool(steveClient *steveV1.Client) (bool, error) {
+	logrus.Info("Checking node pool")
+
+	logrus.Infof("Getting the node using the label [%v]", labelWorker)
+	query, err := url.ParseQuery(labelWorker)
+	if err != nil {
+		return false, err
+	}
+
+	nodeList, err := steveClient.SteveType("node").List(query)
+	if err != nil {
+		return false, err
+	}
+
+	return len(nodeList.Data) >= 3, err
+}
+
+func validateLoadBalancer(client *rancher.Client, clusterID string, steveClient *steveV1.Client, nodePort int, workloadName string) (bool, error) {
+	logrus.Infof("Getting the node using the label [%v]", labelWorker)
+	query, err := url.ParseQuery(labelWorker)
+	if err != nil {
+		return false, err
+	}
+
+	nodeList, err := steveClient.SteveType("node").List(query)
+	if err != nil {
+		return false, err
+	}
+
+	for _, machine := range nodeList.Data {
+		logrus.Info("Getting the node IP")
+		newNode := &corev1.Node{}
+		err = steveV1.ConvertToK8sType(machine.JSONResp, newNode)
+		if err != nil {
+			return false, err
+		}
+
+		nodeIP := kubeapinodes.GetNodeIP(newNode, corev1.NodeExternalIP)
+		if nodeIP == "" {
+			nodeIP = kubeapinodes.GetNodeIP(newNode, corev1.NodeInternalIP)
+		}
+
+		log, err := curlCommand(client, clusterID, fmt.Sprintf("%s:%s/name.html", nodeIP, strconv.Itoa(nodePort)))
+		if err != nil {
+			return false, err
+		}
+
+		return strings.Contains(log, workloadName), err
+	}
+
+	return false, err
+}
+
+func validateHostPortSSH(client *rancher.Client, clusterID string, clusterName string, steveClient *steveV1.Client, hostPort int, workloadName string, namespaceName string) (bool, error) {
```

Comment:
imo, for all of these validates it would make sense to only return an err. And in the case that the log doesn't contain the workload name, just create / throw a new error stating that instead of returning a separate bool.

Reply:
I think that it makes sense, I'll do this refactor